include/aspect/particle/generator/interface.h

```diff
@@ -121,6 +138,15 @@ namespace aspect
           std::pair<types::LevelInd,Particle<dim> >
           generate_particle(const Point<dim> &position,
                             const types::particle_index id) const;
+
+        private:
+          /**
+           * Random number generator and an object that describes a
+           * uniform distribution on the interval [0,1]. These
+           * will be used to generate random particle locations.
+           */
+          boost::mt19937            random_number_generator;
+          boost::uniform_01<double> uniform_distribution_01;
```

Comment:
By the way, these distribution objects have no state and are cheap to create. You may as well do that in the place where you use it.